src/serac/physics/tests/thermal_statics_patch.cpp

```diff
@@ -176,7 +176,7 @@ double solution_error(const ExactSolution& exact_temperature, PatchBoundaryCondi
   HeatTransfer<p, dim> thermal(nonlinear_opts, heat_transfer::direct_linear_options, heat_transfer::default_static_options, "thermal", mesh_tag);
 
   heat_transfer::LinearIsotropicConductor mat(1.0,1.0,1.0);
-  thermal.setMaterial(mat);
+  thermal.setMaterial(std::move(mat));
```

Comment:
I don't like that using `move` has become part of the interface. Why was its use necessary here, but not in the mechanics patch tests? Are the move semantics necessary to meet the objective of this PR, or was it a choice? (Sorry, if this is obvious, I'm a dummy about this topic).

Reply:
I am pretty sure this is just an optimization and is not strictly necessary. It is just telling the compiler that the `mat` object can be "moved" which allows more efficient code.

Reply:
> Why was its use necessary here, but not in the mechanics patch tests? Are the move semantics necessary to meet the objective of this PR, or was it a choice? (Sorry, if this is obvious, I'm a dummy about this topic).

Sadly, C++ makes this pretty counterintuitive. In C++, the spelling `&&` on a function argument represents two fundamentally different ideas, depending on whether the argument is a template parameter of that function or not:

- in `void foo(some_concrete_type && arg)`, arg is "r-value reference" to `some_concrete_type`, which does require the callers of `foo` to use `std::move` when passing in a named variable.

- in `template < typename T > void foo(T && arg)`, arg is a "forwarding reference", which does not require the callers of foo to use `std::move` when passing in a named variable.

The parts of this PR relevant to the original comment are using a mix of pass-by-value and forwarding references, neither of which requires users to `std::move` the arguments, so I'd vote to avoid introducing `std::move`s where they aren't necessary. 

Also, these parts of serac that use forwarding references do so for the wrong reasons (my fault), so it's probably clearer to use the simpler language feature (pass the qfunctions by `const T &`) instead to avoid this kind of confusion in the future, but that's not necessary for this PR.